src/static_allocator/device.rs

```diff
@@ -11,6 +11,13 @@ use std::sync::Arc;
 #[cfg(feature = "allocator_stats")]
 use std::sync::atomic::AtomicUsize;
 
+pub const FREE_MEMORY_SLACK: usize = 1 << 23; // 8 MB
+#[cfg(feature = "recompute")]
+pub const ALLOCATOR_LIMITED_BLOCKS_COUNT: usize = 1340 + 32;
+#[cfg(not(feature = "recompute"))]
+pub const ALLOCATOR_LIMITED_BLOCKS_COUNT: usize = 1695 + 64;
```

Comment:
Why do we fix `ALLOCATOR_LIMITED_BLOCKS_COUNT`?
Iiuc in theory it should depend on `block_size`.
For a given `block_size`, `init_limited` should choose the max number of blocks such that `requested_memory_size_in_bytes <= free_memory_size_in_bytes`, for example
```rust
let blocks_count = free_memory_size_in_bytes / block_size_in_bytes
```

In practice fixing BLOCKS_COUNT is probably fine because `ZKSYNC_DEFAULT_TRACE_LOG_LENGTH` is unlikely to change. I'd be satisfied if we just add
```rust
assert_eq!(ZKSYNC_DEFAULT_TRACE_LOG_LENGTH, 20, "ALLOCATOR_LIMITED_BLOCKS_COUNT may need retuning")
```
somewhere as a reminder to retune block counts if block size changes.

Also, we should probably make sure the production env uses `CUDA_LAZY_LOADING=EAGER`
so we don't hit https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#allocators.
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#cuda-environment-variables says LAZY is already the default on linux :grimacing:

Reply:
> in theory it should depend on block_size and available memory.

hmm, on second thought, my comment assumes our goal is to allocate as many blocks as dram allows. If the goal actually just to allocate as many blocks as we need for any circuit to run, my comment is wrong, and fixing BLOCKS_COUNT is the right thing to do.